habitat/core/batched_env.py

```diff
@@ -15,6 +15,85 @@
 
 import torch  # isort:skip # noqa: F401  must import torch before importing bps_pytorch
 
+from habitat.utils.geometry_utils import  quaternion_rotate_vector 
+from habitat.tasks.utils import cartesian_to_polar
+
+from habitat_sim.utils.common import  quat_from_magnum
+
+
+class StateSensorConfig:
+    def __init__(self, config_key, shape, obs_key):
+        self.config_key = config_key
+        self.shape = shape
+        self.obs_key = obs_key
+    def get_obs(self, state):
+        pass
+
+def _get_spherical_coordinates(source_position, goal_position, source_rotation):
```

Comment:
I'd suggest some kind of unit test for this, even something informal, if you haven't already. And/or step through some cases in the debugger or otherwise make sure you're confident in the correctness. It's hard for me to eyeball correctness.